darts/models/forecasting/baselines.py

```diff
@@ -123,6 +123,53 @@ def predict(self, n: int, num_samples: int = 1, verbose: bool = False):
         return self._build_forecast_series(forecast)
 
 
+class NaiveMovingAverage(LocalForecastingModel):
+    def __init__(self, input_chunk_length: int = 1):
+        """Naive Moving Average Model
+
+        This model forecasts using an auto-regressive moving average (ARMA).
+
+        Parameters
+        ----------
+        input_chunk_length
+            The size of the sliding window used to calculate the moving average
+        """
+        super().__init__()
+        self.input_chunk_length = input_chunk_length
+        self.rolling_window = None
+
+    @property
+    def min_train_series_length(self):
+        return self.input_chunk_length
+
+    def __str__(self):
+        return f"NaiveMovingAverage({self.input_chunk_length})"
+
+    def fit(self, series: TimeSeries):
+        super().fit(series)
+        assert series.n_samples == 1, "This model expects deterministic time series"
```

Comment:
Instead of this assert, it would be cleaner in my opinion to use the `raise_if` method combined with the `is_deterministic` attribute of the `TimeSeries`.